be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -21,7 +21,19 @@ class ORCHdfsFileStream : public orc::InputStream {
 
     uint64_t getLength() const override { return _length; }
 
-    uint64_t getNaturalReadSize() const override { return 2 * 1024 * 1024; }
+    uint64_t getNaturalReadSize() const override { return 1 * 1024 * 1024; }
+
+    // It's 1/4 of NaturalReadSize. It's for read size after doing seek.
+    // When doing read after seek, we make assumption that we are doing random read because of seeking row group.
+    // And if we still use NaturalReadSize we probably read many row groups
+    // after the row group we want to read, and that will amplify read IO bytes.
+
+    // So the best way is to reduce read size, hopefully we just read that row group in one shot.
+    // We also have chance that we may not read enough at this shot, then we fallback to NaturalReadSize to read.
+    // The cost is, there is a extra IO, and we read 1/4 of NaturalReadSize more data.
+    // And the potential gain is, we save 3/4 of NaturalReadSize IO bytes.
+
+    uint64_t getNaturalReadSizeAfterSeek() const override { return 256 * 1024; }
```

Comment:
why chose quarter of NaturalReadSize?  can 64KB, 32KB be used instead?

Reply:
Because 256KB can usually cover a row group of column. It can not be too small. If it's too small, a reading can not cover a row group of column and we fallback to normal read, that means we pay the cost of a extra reading.